lib/safe_json_serializer.rb

```diff
@@ -0,0 +1,20 @@
+module SafeJSONSerializer
+  def load(source, proc = nil, options = {})
+    # we use parse, because load is unsafe
+    json = JSON.parse(source, quirks_mode: true)
+    if json.is_a?(Hash) && json['flash']
+      flash = ActionDispatch::Flash::FlashHash.new
+      flash.update(json.delete('flash').to_h.with_indifferent_access.symbolize_keys)
+      json['flash'] = flash
+    end
+    json
+  end
+
+  module_function :load
+
+  def dump(obj, anIO = nil, limit = nil)
+    obj.to_json
```

Comment:
maybe use: 

```ruby
JSON.generate(obj, quirks_mode: true)
```

similar to what was used in rails: [json_serializer](https://github.com/rails/rails/pull/13692/files#diff-cf6d7cf861425a8fad71dbbbeb70f739R9)

Reply:
Done